created more extra deps (#4662)
sunpy/sunpy: docs/guide/installation.rst

To install the Miniconda Python distribution follow the instructions at
`here <https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html>`__.
Although Miniconda makes it simple to switch between Python versions, we recommend that new users install
the latest Python 3.x version of Miniconda as SunPy only supports Python 3.7+.

The reason we choose Miniconda over Anaconda, is mainly due to the size as Anaconda comes with a full install of packages you probably do not need and this way you have more direct control over what has been installed into your Python virtual environment.
Furthermore, you bypass the need for the conda resolver to sort out your root environment which should make "conda" faster to use.

Installing sunpy using Miniconda
--------------------------------

To install SunPy launch a system command prompt or the 'Anaconda Prompt' (under Windows).
First configure conda for to add the `conda-forge channel <https://conda-forge.org/>__`::

    conda config --add channels conda-forge
    conda config --set channel_priority strict

and now to install sunpy::

>     conda install sunpy

Reviewer: Should we be recommending people start a fresh env for sunpy? Is that too far for the basic install docs?

Author: I thought this was aimed at people without a previous sunpy install. Who are installing it for the first time.

Reviewer: Sure, I just mean changing this to `conda create -n sunpy python sunpy; conda activate sunpy`

Reviewer: (I am not sure this is a good idea, but in general using a venv is always good right?)

Author: I will redraft this section further then.

Author: Fixed.

Reviewer: I would be interested to know other peoples opinions on this, and we might need to make it clearer that you have to activate the env each time. Then does our install docs just become a conda guide :exploding_head: 

Maybe we should actually just give people the install into the root env command and link them to a environments guide?

Reviewer 2: I definitely think its good to do a 'best practice' but for a lot of users, however we should probably be aware that many scientific users, especially those not fully accustomed to environments etc, and basically the majority of the coding will use sunpy etc and just want all the required packages in their base env we should probably have a simple method for doing this.

> and now to install sunpy within a the default conda virtual environment::

Reviewer 3: Remove "a" before "the"

This will install sunpy and every package it needs to function.

.. note::
    We strongly recommend using a `Python virtual environment <https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/>`__ or a `conda virtual environment. <https://towardsdatascience.com/getting-started-with-python-environments-using-conda-32e9f2779307>`__

Updating sunpy
--------------

You can update to the latest version by running::

    conda update sunpy

Installing sunpy on top of an existing scientific Python environment
--------------------------------------------------------------------

Conda
^^^^^

If you want to install sunpy within a pre-existing conda environment, you will want to activate the virtual environment and run::

>     $ conda install sunpy

Reviewer 3: Maybe add the `activate` command above this?

This assumes you have the conda-forge channel added (as above).

Pip
^^^

This is for installing sunpy within a scientific Python distribution or environment, where ``pip`` has been used to install packages.

To acquire a fully working sunpy installation, simply run::

    pip install sunpy[all]

.. note::
    If this does not work, it could be due to a missing C compiler (e.g., ``gcc`` or ``clang``) that is required to build sunpy at install.

>     Getting the compiler either from your system package manager or XCode or Anaconda should address this.

Reviewer: ```suggestion
    Getting the compiler either from your system package manager, XCode or Anaconda should address this.
```

>     You can use Conda as a virtual environment manager instead of one of the more common Python virtual environment managers.

Reviewer: This is really a recommendation if you don't already have python setup, and it's in a section which assumes you do. It feels a little out of place?

Author: Fixed.

> If you have a reason to bypass this, you can sunpy with no optional dependencies::

Reviewer: ```suggestion
If you have a reason to want a more minimal installation, you can install sunpy with no optional dependencies, however this means a lot of submodules will not import::
```

    pip install sunpy

> To install sunpy with map- and timeseries-based dependencies::

Reviewer: ```suggestion
It is possible to select which "extra" dependencies you want to install, if you know you only need certain submodules.
For example to install sunpy with map- and timeseries-based dependencies run::
```

> This ensure that your version of sunpy is fully operational.

Reviewer: While this is true, the extras are really for downstream packages. Should we adjust this wording to put "to get a fully working sunpy installation run: `pip install sunpy[all]`` first and then "if for some reason you only want to install the requirements for some parts of sunpy you can do `xxx`" afterwards?

i.e. making the path to the recommended thing as short as simple as possible.

    pip install sunpy[map,timeseries]

The available options are: ``[asdf]``, ``[dask]``,``[database]``,``[instr]``,``[image]``,``[jpeg2000]``,``[map]``,``[net]``,``[timeseries]``,``[visualization]``.

If you want to develop sunpy we would strongly recommend reading the `Newcomers' Guide <https://docs.sunpy.org/en/latest/dev_guide/newcomers.html>`__.

.. note::
    If you get a ``PermissionError`` this means that you do not have the required administrative access to install new packages to your Python installation.

    Do **not** install sunpy or other third-party packages using ``sudo``.

>     This error implies you either an incorrectly configured virtual environment or it was not activate.

Reviewer 3: "are either in" and "activated"

>     In this case you may consider using the ``--user`` option to install the package into your home directory.

Reviewer: I would probably recommend venv here before `--user` saying something like "if you really don't want to use venv, you can always use `--user`".

Author: Fixed.

>     Do **not** install sunpy or other third-party packages using ``sudo``.

Reviewer: I would put this on the second line.

Author: Fixed.

************
Installation
************

Requirements
============

These are the minimum versions of packages needed to install sunpy:

- python 3.7
- astropy 4.0.3

> - scipy 1.2

Reviewer 4: ```suggestion
```

    conda config --set channel_priority strict

and now to install sunpy within a new virtual environment::

    $ conda create -n sunpy python sunpy
    $ conda activate sunpy

This will install sunpy and every package it needs to function.

Updating sunpy
--------------

You can update to the latest version by running::

    conda update sunpy

Installing sunpy on top of an existing scientific Python environment
--------------------------------------------------------------------

Conda
^^^^^

If you want to install sunpy within a pre-existing conda environment, you can run::

    $ conda install sunpy

This assumes you have the conda-forge channel added (as above).

Pip
^^^

This is for installing sunpy within a scientific Python distribution, where ``pip`` has been used to install  packages.

We do provide compiled binaries for sunpy for Linux and Mac OS X (we do not compile our C extension on Windows), so you might not need a C compiler.
If there is no compiled binary, you will need a C compiler (e.g., ``gcc`` or ``clang``) to be installed as we have a C library within sunpy that is built at install time.
If you use Miniconda, you can get these compilers from there.

> On Linux, using the package manager for your distribution will usually be the easiest route, while on MacOS X you will need the XCode command line tools.

Reviewer: I wonder if this should also be moved after the install instructions? We could say something like "if it doesn't work then you might need a C compiler"?
